lib/ansible/module_utils/common/text/converters.py

```diff
@@ -23,7 +26,27 @@
                                       'surrogate_then_replace'))
 
 
-def to_bytes(obj, encoding='utf-8', errors=None, nonstring='simplerepr'):
+@_t.overload
+def to_bytes(
+    obj: _T,
+    encoding: str = 'utf-8',
+    errors: str | None = None,
+    *,
+    nonstring: _t.Literal['passthru'],
+) -> _T | bytes: ...
+
+
+@_t.overload
+def to_bytes(
+    obj: _t.Any,
+    encoding: str = 'utf-8',
+    errors: str | None = None,
+    *,
+    nonstring: _t.Literal['simplerepr', 'empty', 'strict'] = 'simplerepr',
+) -> bytes: ...
+
+
+def to_bytes(obj, encoding='utf-8', errors=None, *, nonstring='simplerepr'):
```

Comment:
This should still have annotations. I think some tooling will get confused if you don't add it. Not sure if MyPy even checks this function if you don't.

Reply:
> This should still have annotations. I think some tooling will get confused if you don't add it. Not sure if MyPy even checks this function if you don't.
mypy does seem to handle it fine, but I'll look.

> Is the keyword-only API change intentional? This makes the PR not just typing-scoped but also a feature-level change.
Also yes, it was intentional, mypy actually didn't detect certain scenarios as invalid during my testing until I add the keyword only change. But fwiw, I did evaluate all uses and found no use that wasn't already using `nonstring` as a keyword, both in core and all collections.

Reply:
I've added type hints, as it was easier than research. Also I didn't add explicit Literals for all of the errors on purpose, as I really don't want to own the list.